walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,76 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long workingPeriod = request.getDuration();
+    int totalHours = 24*60;
+    // If meeting duration is greater than a day, then meeting cannot happen.
+    if (workingPeriod > TimeRange.WHOLE_DAY.duration()){ return new ArrayList<TimeRange>();}
```

Comment:
```suggestion
    if (workingPeriod > TimeRange.WHOLE_DAY.duration()) return new ArrayList<TimeRange>();
```

it is more consistent with line 34